src/snowflake/cli/plugins/git/commands.py

```diff
@@ -64,3 +69,69 @@ def fetch(
     **options,
 ) -> CommandResult:
     return QueryResult(GitManager().fetch(repo_name=repository_name))
+
+
+@app.command(
+    "copy",
+    help="Copies all files from given state of repository to local directory or stage.",
+    requires_connection=True,
+)
+def copy(
+    repository_path: str = RepoPathArgument,
+    destination_path: str = typer.Argument(
+        help="Target path for copy operation. Should be stage or local file path.",
+    ),
+    parallel: int = typer.Option(
+        4,
+        help="Number of parallel threads to use when downloading files.",
+    ),
+    **options,
+):
+    _assert_repository_path_is_stage("REPOSITORY_PATH", path=repository_path)
+    is_copy = is_stage_path(destination_path)
+    if is_copy:
+        cursor = GitManager().copy(
+            repo_path=repository_path, destination_path=destination_path
+        )
+    else:
+        target = Path(destination_path).resolve()
+        cursor = GitManager().get(
+            repo_path=repository_path, target_path=target, parallel=parallel
+        )
+    return QueryResult(cursor)
+
+
+@app.command(
+    "setup",
+    help="Create a git repository stage.",
+    requires_connection=True,
+)
+def setup(
+    repo_name: str = RepoNameArgument,
+    api_integration: str = typer.Option(
+        ...,
+        "--api-integration",
+        "--api",
+        help="identifier of the api integration object to use",
+    ),
+    url=typer.Option(default=..., help="url to clone git repository from"),
```

Comment:
1. IMO all help messages should start with a capitilized letter.
2. Can we specify parameter type for `url` (probably `str`)?